bindings/nodejs/inc/nodejs_common.h

```diff
@@ -140,16 +142,17 @@ struct NODEJS_MODULE_HANDLE_DATA
         start_pending = isPending;
     }
 
-    BROKER_HANDLE               broker;
-    std::string                 main_path;
-    std::string                 configuration_json;
-    v8::Isolate                 *v8_isolate;
-    v8::Persistent<v8::Object>  module_object;
-    size_t                      module_id;
-    PFNMODULE_START             on_module_start;
-    NodeModuleState             module_state;
-    bool                        start_pending;
-    nodejs_module::Lock         object_lock;
+    BROKER_HANDLE broker;
+    std::promise<NodeModuleState> create_complete;
+    std::string main_path;
+    std::string configuration_json;
+    v8::Isolate *v8_isolate;
+    v8::Persistent<v8::Object> module_object;
+    size_t module_id;
+    PFNMODULE_START on_module_start;
+    NodeModuleState module_state;
+    bool start_pending;
```

Comment:
The idea behind having this flag `start_pending` was to handle the case where a call to `NODEJS_Start` occurs before the module's state has transitioned to `initialized` from `initializing`. Now that `NODEJS_Create` is a blocking call and we no longer have an `initializing` state, we may not need this flag anymore. Consider removing this field and the associated logic.